akka-persistence-typed-tests/src/test/java/jdocs/akka/persistence/typed/ReplicatedAuctionExampleTest.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.akka.persistence.typed;
+
+import akka.actor.testkit.typed.javadsl.LogCapturing;
+import akka.actor.testkit.typed.javadsl.TestKitJunitResource;
+import akka.actor.testkit.typed.javadsl.TestProbe;
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.persistence.testkit.PersistenceTestKitPlugin;
+import akka.persistence.testkit.query.scaladsl.PersistenceTestKitReadJournal;
+import akka.persistence.typed.ReplicaId;
+import akka.persistence.typed.javadsl.*;
+import akka.serialization.jackson.CborSerializable;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.scalatestplus.junit.JUnitSuite;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static jdocs.akka.persistence.typed.ReplicatedAuctionExample.*;
+import static org.junit.Assert.assertEquals;
+
+public class ReplicatedAuctionExampleTest extends JUnitSuite {
+  @ClassRule
+  public static final TestKitJunitResource testKit =
+      new TestKitJunitResource(PersistenceTestKitPlugin.getInstance().config());
+
+  @Rule public final LogCapturing logCapturing = new LogCapturing();
+
+  @Test
+  public void auctionExample() {
+    AuctionSetup setupA =
+        new AuctionSetup(
+            "old-skis",
+            new Bid("chbatey", 12, Instant.now(), R1),
+            Instant.now().plusSeconds(10),
+            true);
+
+    AuctionSetup setupB =
+        new AuctionSetup(
+            "old-skis",
+            new Bid("chbatey", 12, Instant.now(), R1),
+            Instant.now().plusSeconds(10),
+            false);
+
+    ActorRef<Command> replicaA = testKit.spawn(create(setupA, R1));
+    ActorRef<Command> replicaB = testKit.spawn(create(setupA, R2));
+
+    replicaA.tell(new OfferBid("me", 100));
+    replicaA.tell(new OfferBid("me", 99));
+    replicaA.tell(new OfferBid("me", 202));
+
+    TestProbe<Bid> replyProbe = testKit.createTestProbe();
+    replyProbe.awaitAssert(
+        () -> {
+          replicaA.tell(new GetHighestBid(replyProbe.ref()));
+          Bid bid = replyProbe.expectMessageClass(Bid.class);
+          assertEquals(bid.offer, 202);
+          return bid;
+        });
+
+    replicaA.tell(Finish.INSTANCE);
+
+    TestProbe<Boolean> finishProbe = testKit.createTestProbe();
+    finishProbe.awaitAssert(
+        () -> {
+          replicaA.tell(new IsClosed(finishProbe.ref()));
+          return finishProbe.expectMessage(true);
+        });
+    finishProbe.awaitAssert(
+        () -> {
+          replicaB.tell(new IsClosed(finishProbe.ref()));
+          return finishProbe.expectMessage(true);
+        });
+  }
+}
+
+class ReplicatedAuctionExample
```

Comment:
Hmm, you can put multiple classes in same file in java if they are not public or what is this dark magic?